packages/datadog-plugin-graphql/src/resolve.js

```diff
@@ -15,7 +15,12 @@ class GraphQLResolvePlugin extends Plugin {
     this.addSub('apm:graphql:resolve:start', ({ info, context }) => {
       const store = storage.getStore()
       depthPredicate(info, this.config, (computedPath) => {
-        if (!hasLikePath(context, computedPath)) {
+        const computedPathString = computedPath.join('.')
+        if ((!this.config.collapse && !context.fields[computedPathString]) ||
+             !context.collapsedPaths[computedPathString]) {
+          // cache the collapsed string here
+          if (this.config.collapse) context.collapsedPaths[computedPathString] = true
```

Comment:
If this is done on the subscriber side it shouldn't need to configure the publisher.